service/frontend/nexus_handler.go

```diff
@@ -313,7 +397,65 @@ func (h *nexusHandler) CancelOperation(ctx context.Context, operation, id string
 	return fmt.Errorf("unhandled response outcome: %T", response.GetOutcome()) //nolint:goerr113
 }
 
-// convertNexusHandlerError converts any 5xx user handler error to a downsream error.
+func (h *nexusHandler) forwardCancelOperation(ctx context.Context, operation string, id string, options nexus.CancelOperationOptions, oc *operationContext) error {
+	options.Header = setHeadersForForwardedRequest(options.Header)
+
+	client, err := h.nexusClientForActiveCluster(oc)
+	if err != nil {
+		return err
+	}
+
+	handle, err := client.NewHandle(operation, id)
+	if err != nil {
+		oc.logger.Warn("invalid Nexus cancel operation.", tag.Error(err))
+		return nexus.HandlerErrorf(nexus.HandlerErrorTypeBadRequest, "invalid operation")
+	}
+
+	err = handle.Cancel(ctx, options)
+	if err != nil {
+		oc.logger.Error("received error from remote cluster for forwarded Nexus cancel operation request.", tag.Error(err))
+		oc.metricsHandler = oc.metricsHandler.WithTags(metrics.NexusOutcomeTag("forwarded_request_error"))
+		return commonnexus.ConvertClientError(err)
+	}
+
+	return nil
+}
+
+func setHeadersForForwardedRequest(headers nexus.Header) nexus.Header {
```

Comment:
I _think_ header will never be nil on the server side, if that's the case you can drop this whole function and set the header inline.